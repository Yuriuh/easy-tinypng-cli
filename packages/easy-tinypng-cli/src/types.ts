export interface Config {
  targetDir: string
}

export interface ExportConfig {
  configs: Array<Config>
  APIKey: string
}
