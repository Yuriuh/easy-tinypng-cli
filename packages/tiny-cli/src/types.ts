export interface Config {
  targetDir: string
}

export interface ExportConfig {
  configs: Array<Config>
}
