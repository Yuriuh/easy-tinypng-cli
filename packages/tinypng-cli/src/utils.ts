import type { ExportConfig } from './types'
function defineTinyConfig(config: ExportConfig): ExportConfig {
  return config
}

export {
  defineTinyConfig,
}
