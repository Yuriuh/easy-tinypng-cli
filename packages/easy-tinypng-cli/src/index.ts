import { loadConfig } from 'unconfig'
import type { ExportConfig } from './types'
import { getAllConfigs, startOptimize } from './utils.js'
const { config } = await loadConfig<ExportConfig>({
  sources: [
    {
      files: 'tiny.config',
      extensions: ['ts', 'mts', 'cts', 'js', 'mjs', 'cjs', 'json', ''],
    },
  ],
})

const configs = getAllConfigs(config)
startOptimize(configs)

