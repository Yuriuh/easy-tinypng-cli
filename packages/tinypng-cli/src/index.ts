import { loadConfig } from 'unconfig'
import type { ExportConfig } from './types'
import { getAllConfigs, startOptimize } from './utils'
const { config } = await loadConfig<ExportConfig>({
  sources: [
    {
      files: 'tiny.config',
      extensions: ['ts', 'mts', 'cts', 'js', 'mjs', 'cjs', 'json', ''],
    },
  ],
})
// 获取config
const configs = getAllConfigs(config)
// 监听targetDir
startOptimize(configs)

