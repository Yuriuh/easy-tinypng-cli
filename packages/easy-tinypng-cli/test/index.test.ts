import { cwd } from 'node:process'
import path from 'path'
import fs from 'node:fs'
import { describe, expect, it } from 'vitest'
import fse from 'fs-extra'
import ora from 'ora'
import { isRecord, record } from '../src/utils'
describe('should', () => {
  it('exported', () => {
    expect(1).toEqual(1)
  })
})

it('isFileExist', async () => {
  function isFileExist(pathDir: string) {
    console.warn(pathDir, '======================================')
    return new Promise((resolve) => {
      fs.access(pathDir, fs.constants.F_OK, (err) => {
        resolve(!err)
      })
    })
  }

  const result = await isFileExist(path.resolve(cwd(), 'src/react.png'))
  expect(result).toEqual(true)
})

it('test jsonRead', () => {
  const jsonPathDir = path.resolve(cwd(), '../src/hello.json')
  const result = fse.readJSONSync(jsonPathDir)
  expect(result).toMatchSnapshot()
})

it('test pathName exist', () => {
  const pathName = path.resolve(cwd(), '../src/hello.json')
  const fileName = path.basename(pathName)
  expect(fileName).toEqual('hello.json')
})

it('test writeJsonFile', () => {
  const pathDir = path.resolve(cwd(), '../src/C.png')
  record(pathDir)
})

it('get extname', () => {
  const pathDir = path.resolve(cwd(), './src')
  const extname = path.extname(pathDir).slice(1)
  expect(extname).toEqual('text')
})

it('is record', async () => {
  const pathDir = path.resolve(cwd(), '../src/B.png')
  const recorded = await isRecord(pathDir)
  expect(recorded).toEqual(true)
})

it('test spinner', () => {
  const spinner = ora('loading').start()
  setTimeout(() => {
    spinner.color = 'yellow'
    spinner.text = 'Loading rainbows'
  }, 1000)
})
