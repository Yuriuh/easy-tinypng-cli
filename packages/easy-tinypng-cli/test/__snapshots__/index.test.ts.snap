// Vitest Snapshot v1

exports[`isFileExist 1`] = `undefined`;

exports[`test jsonRead 1`] = `
{
  "uname": "sudongyuer",
}
`;
